java/src/jmri/jmrit/display/layoutEditor/TrackSegmentView.java

```diff
@@ -564,6 +564,15 @@ public Rectangle2D getBounds() {
         result = new Rectangle2D.Double(ep1.getX(), ep1.getY(), 0, 0);
         result.add(ep2);
 
+        if (isCircle()) {
+            result.add(getCoordsCenterCircle());
+        } else if (isBezier()) {
+            for (int index = 0; index < bezierControlPoints.size(); index++) {
+                result.add(bezierControlPoints.get(index));
+            }
+        }
+        result.add(getCentreSeg());
```

Comment:
The infinite loop that @danielb987 saw in CI goes through here.  Line 1909 of TrackSegment view in `getCentreSeg` does a `setCoordsCenter` call, which goes through getBounds, which in line 574 calls `getCentreSeg` again.  Not sure whether the underlying problem is here or in line 1909 (why is a `get` call _setting_ something?), but this is the new code.

Reply:
Thanks for the catch/analysis!

I added a getCentreSeg() call to getBounds() method to add that point to the returned bounds.
But (as you said) the getCentreSeg() method uses getBounds() to calculate where the center seg should be… 
So we can't call getCentreSeg() here… but since it calls super.setCoordsCenter(…) we can use super.getCoordsCenter to get that point (without re-calculating it) instead. So… fixed(?)
 ;-)